lms/views/admin/email.py

```diff
@@ -0,0 +1,51 @@
+from pyramid.httpexceptions import HTTPBadRequest, HTTPFound
+from pyramid.view import view_config, view_defaults
+
+from lms.security import Permissions
+from lms.tasks.email_digests import send_instructor_email_digests
+
+
+@view_defaults(route_name="admin.email", permission=Permissions.ADMIN)
+class AdminEmailViews:
+    def __init__(self, request):
+        self.request = request
+
+    @view_config(
+        route_name="admin.email",
+        request_method="GET",
+        renderer="lms:templates/admin/email.html.jinja2",
+    )
+    def get(self):
+        return {}
+
+    @view_config(route_name="admin.email", request_method="POST")
+    def post(self):
+        to_email = self.request.POST["to_email"].strip()
+        h_userids = self.request.POST["h_userids"].strip().split()
+        since = self.request.POST["since"].strip()
+        until = self.request.POST["until"].strip()
+
+        if not to_email:
+            raise HTTPBadRequest(
+                "You must enter an email address to send the test email(s) to."
+            )
+
+        if not to_email.endswith("@hypothes.is"):
+            raise HTTPBadRequest(
+                "Test emails can only be sent to @hypothes.is addresses."
+            )
+
+        if len(h_userids) > 3:
```

Comment:
Maybe you can add this later but in the same way we limit this here we should:

- Validate that dates are present and are actual parseable dates
- Validate that the date range is somewhat constrain, if 3 users is the limit maybe a week or so should also be the limit.

Reply:
Added these extra validation checks: https://github.com/hypothesis/lms/pull/5130/files#diff-6085d93aa772ea75f9371cc9ad76ecc08a66b296511f2a424be63ab7044605bcR45-R61

I also collapsed the invalid data tests into a single parametrized test: https://github.com/hypothesis/lms/pull/5130/files#diff-6085d93aa772ea75f9371cc9ad76ecc08a66b296511f2a424be63ab7044605bcR45-R61